FT2 - Link react components with backend server

pablodavid97 opened the pull request:
Add
- NoteList react component to make request to server and retrieve
note data

Modify
- Project structure so that server is the starting point of the app
- Application PORTS so that react runs on port 3000 and server runs on
port 8080.
- Env file to include additional environment variables
- index.js to include cors module

Commit:
FT2 - Link react components with backend server

Add
- NoteList react component to make request to server and retrieve
note data

Modify
- Project structure so that server is the starting point of the app
- Application PORTS so that react runs on port 3000 and server runs on
port 8080.
- Env file to include additional environment variables
- index.js to include cors module

moejoe1996 commented on react/src/components/Note.jsx:
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 
-function Note() {
+function Note(props) {
```
More properties should be passed here (e.x: `backgoundColor`, `created_at`). But they can be added as we build the `Post` functionality.

pablodavid97 replied:
Yes, we can add them in future tickets. I think it is fine for now.

moejoe1996 commented on react/src/components/NoteList.jsx:
```diff
@@ -0,0 +1,23 @@
+
+function List(){
+ const [list, setList] = useState([]);
+ const url = 'http://localhost:8080/api/cards';
```
We should have a `config` file to store this variables. To avoid repeating ourselves. When we need to make a call to the API we can simply do `constant url = '{host}/cards'`

pablodavid97 replied:
Good idea! I'll create config file and reach out when changes are ready.

moejoe1996 reviewed (commented):
Comments added

Commit:
FT2 - Create config files to store credentials

Create config files for reusability purposes both for the backend and
front end sides.

Add:
- config/default.js file for server variables
- config.json file for react variables

Modify:
- NoteList.jsx to reference host via config file
- index.js to include variables from config file
- cards_controller.js to include api host and port using config file
- .env.example file to only include server variables

pablodavid97 commented on config/default.js:
```diff
@@ -0,0 +1,17 @@
+require('dotenv').config();
+
```
@moejoe1996 created config file

moejoe1996 commented on react/src/ config.json:
```diff
@@ -0,0 +1,3 @@
+{
```
This file seems to have a whitespace in the name `react/src/ config.json`

pablodavid97 replied:
ok

moejoe1996 commented on react/src/components/NoteList.jsx:
```diff
@@ -0,0 +1,24 @@
+import React, {useEffect, useState} from 'react';
+import Note from './Note';
+import configData from '../ config.json';
```
whitespace in filepath

pablodavid97 replied:
ok

moejoe1996 commented on src/controller/cards_controller.js:
```diff
@@ -1,4 +1,9 @@
+const config = require('config');
+
 const cardsController = (Card) => {
+    const host = config.get('server.host');
```
is it posible to use the deconstruct here? `const { host, port } = config.get('server');`

pablodavid97 replied:
yes it is, I'll make the changes

moejoe1996 reviewed (commented):
comments added

moejoe1996 reviewed (approved):
LGTM!
